### PR TITLE
BI-1167 - Germplasm Import - Validate Required Fields

### PR DESCRIPTION
**Pull request by nickpalladino**
# Description
**Story:** [BI-1167](https://breedinginsight.atlassian.net/browse/BI-1167?atlOrigin=eyJpIjoiNDRjM2Y3ZjYzMDZiNDBmNWFhMzY2ZjAyNDcyM2UwMjMiLCJwIjoiaiJ9)

- Added error list display for first stage errors

# Dependencies
- BI-1167 bi-api

# Testing
- Create a germplasm import files with various combinations of required fields missing and present to see that errors shown are as expected.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have create/modified unit tests to cover this change
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [x] I have run TAF.

### CODE REVIEW

**Review by timparsons — CHANGES_REQUESTED** [review]
Ran the following files through:

- Valid file - PASSED
- Missing columns - Error shows in the banner, not in the error section, but there's already a story for this ([BI-1324](https://breedinginsight.atlassian.net/browse/BI-1324))
- Missing required data - PASSED
- Column names in different cases - FAILED - file -> [germplasm_column_casing.xls](https://github.com/Breeding-Insight/bi-web/files/8064404/germplasm_column_casing.xls)

File: src/views/import/ImportTemplate.vue [hunk 1]
```diff
@@ -335,7 +335,13 @@ export default class ImportTemplate extends ProgramsBase {
       }
       // this.importService.send(ImportEvent.IMPORT_SUCCESS) is in getDataUpload()
     } catch(e) {
-      if (e.response && e.response.status == 422 && e.response.statusText) {
+      if (e.response && e.response.status == 422 && e.response.data && e.response.data.rowErrors) {
+        this.import_errors = ValidationErrorService.parseError(e);
+        if(this.import_errors==null) {
```
Comment (ctucker3) [inline src/views/import/ImportTemplate.vue:340]: Should this be an `undefined` check? I don't see anywhere in the `parseError` method where null is returned.

File: src/views/import/ImportTemplate.vue [hunk 2]
```diff
@@ -335,7 +335,13 @@ export default class ImportTemplate extends ProgramsBase {
...
+          this.$emit('show-error-notification', `Errors: ${e.response!.progress!.message!}`);
```
Comment (ctucker3) [inline src/views/import/ImportTemplate.vue:341]: What if there is no error message? (not sure if this case can ever happen)

**Review by ctucker3 — CHANGES_REQUESTED** [review]
Overall looks good! Just a few small questions/requests.

File: src/views/import/ImportTemplate.vue (hunk 1 above)
Comment (nickpalladino) [reply src/views/import/ImportTemplate.vue:340]: Good point, this was copied over from a previous section that did have a null being returned. Maybe it doesn't need any check and the FileSelectMessageBox component can just display unknown error when import_errors doesn't have any validation errors?

File: src/views/import/ImportTemplate.vue (hunk 2 above)
Comment (nickpalladino) [reply src/views/import/ImportTemplate.vue:341]: Based on the previous comment I think maybe this check should just be removed.

File: src/views/import/ImportTemplate.vue (hunk 1 above)
Comment (ctucker3) [reply src/views/import/ImportTemplate.vue:340]: Sounds good to me!

File: src/views/import/ImportTemplate.vue (hunk 2 above)
Comment (ctucker3) [reply src/views/import/ImportTemplate.vue:341]: Sounds good!

File: src/views/import/ImportTemplate.vue (hunk 1 above)
Comment (nickpalladino) [reply src/views/import/ImportTemplate.vue]: Check has been removed.

**Review by timparsons — APPROVED** [review]

**Review by ctucker3 — APPROVED** [review]

**Pull request merged**